app/controlplane/pkg/biz/organization.go

```diff
@@ -244,3 +248,79 @@ func (uc *OrganizationUseCase) Delete(ctx context.Context, id string) error {
 	// Delete the organization
 	return uc.orgRepo.Delete(ctx, orgUUID)
 }
+
+// AutoOnboardOrganizations creates the organizations specified in the onboarding config and assigns the user to them
+// with the specified role if they are not already a member.
+func (uc *OrganizationUseCase) AutoOnboardOrganizations(ctx context.Context, userID string) error {
+	// Parse user UUID
+	userUUID, err := uuid.Parse(userID)
+	if err != nil {
+		return NewErrInvalidUUID(err)
+	}
+
+	for _, spec := range uc.onboardingConfig {
+		// Ensure the organization exists or create it if it doesn't
+		org, err := uc.ensureOrganizationExists(ctx, spec)
+		if err != nil {
+			return fmt.Errorf("failed to ensure organization exists: %w", err)
+		}
+
+		// Parse organization UUID
+		orgUUID, err := uuid.Parse(org.ID)
+		if err != nil {
+			return NewErrInvalidUUID(err)
+		}
+
+		// Ensure user membership
+		if err := uc.ensureUserMembership(ctx, orgUUID, userUUID, spec); err != nil {
+			return fmt.Errorf("failed to ensure user membership: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// ensureOrganizationExists ensures that an organization specified by the onboarding configuration exists.
+// If the organization does not exist, it creates it.
+func (uc *OrganizationUseCase) ensureOrganizationExists(ctx context.Context, spec *conf.OnboardingSpec) (*Organization, error) {
```

Comment:
Not a blocker but in general I prefer to make the arguments as scooped down as possible. For example instead of the whole spec,  here I think you just need the name string, and in the other one you need the role.

Reply:
👍  I agree. It would be simpler to just pass the name of the org.

Reply:
I agree with you but in this case, we need the whole spec, the expected roles as well, so effectively what's being passed in the name of the orgs and the roles it should be applied.